lib/TPP/ConvertLinalgToXsmm.cpp

```diff
@@ -720,6 +725,81 @@ makeMinorDimensionsInnerMost(RewriterBase &rewriter, linalg::GenericOp linalgOp,
   return linalgOp;
 }
 
+template <typename OpTy>
+static LogicalResult foldFlags(RewriterBase &rewriter, OpTy gemmDispatchOp) {
+  SmallVector<Operation *> users(gemmDispatchOp.getResults().getUsers());
+  if (users.size() != 1)
+    return failure();
```

Comment:
Wouldn't it be more versatile to start from invokes, look at their operands, and update their dispatch if they happen to be zero initialized?

Reply:
If we use pattern rewriter, the root op needs to be either updated in-place, replaced, or erased (https://mlir.llvm.org/docs/PatternRewriter/#restrictions); since I don't touch the invoke, this would
not be the correct usage of pattern rewriter.

Reply:
That's fair but then we could use a different tool.
Anyway, it's just a general thought. If it is fine as is, then no changes needed.